document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -132,6 +134,44 @@ public boolean upsert(Key key, Document document) throws IOException {
     }
   }
 
+  /**
+   * Bulk updates existing documents if condition for the corresponding document evaluates to true.
+   */
+  @Override
+  public boolean bulkUpdate(List<Triple<Key, Document, Filter>> documents) {
+    try {
+      BulkWriteResult result = bulkUpdateImpl(documents);
+      LOGGER.debug(result.toString());
```

Comment:
can you wrap this inside if debugEnabled?

Reply:
I think this should be fine, we are not doing this in other places.

Reply:
done